controllers/application_controller.go

```diff
@@ -128,7 +128,11 @@ func (r *ApplicationReconciler) fetchComponentListResources(ctx context.Context,
 
 		list := &unstructured.UnstructuredList{}
 		list.SetGroupVersionKind(mapping.GroupVersionKind)
-		if err = r.Client.List(ctx, list, client.InNamespace(namespace), client.MatchingLabels(selector.MatchLabels)); err != nil {
+		var listOptions = client.MatchingLabels{}
```

Comment:
Is `selector.MatchExpressions` not supported?

Reply:
It is not supported in the current version. We can open a new issue to add support for it.